start-tests.sh

```diff
@@ -7,106 +7,267 @@
-  export CYPRESS_OC_IDP=kube:admin
+# Create dir for kube configs.
+mkdir -p ./kube/config
+
+# Check to see if OPTIONS_HUB_BASEDOMAIN, OPTIONS_HUB_USER, or OPTIONS_HUB_PASSWORD are missing. We need these to run the UI test.
+if [[ -z $OPTIONS_HUB_BASEDOMAIN || -z $OPTIONS_HUB_USER || -z $OPTIONS_HUB_PASSWORD ]]; then
+  log_color "red" "One or more exported variables are undefined for hub cluster." "(set ${PURPLE}OPTIONS_HUB_BASEDOMAIN, OPTIONS_HUB_BASEDOMAIN, and OPTIONS_HUB_BASEDOMAIN${NC} to execute the test with environment variables)\n"
+
+  # Check to see if the kubeconfig for the hub cluster is available.
+  if [[ ! -f $OPTIONS_HUB_KUBECONFIG ]]; then
+    log_color "red" "The kubeconfig file for the hub cluster was not located." "(set ${PURPLE}KUBECONFIG${NC} to ${YELLOW}$OPTIONS_HUB_KUBECONFIG${NC} and oc login to create kubeconfig file)"
+    exit 1
+  else
+    # To run the E2E test, we need the password to log into the UI. If it's not exported or available from the options.yaml file, skip the UI test.
+    if [[ -z $OPTIONS_HUB_PASSWORD || "$OPTIONS_HUB_PASSWORD" == "null" ]]; then
+      log_color "purple" "OPTIONS_HUB_PASSWORD" "is required to login into the ACM console. ${YELLOW}Skipping UI test.${NC}\n"
+      export SKIP_UI_TEST=true
+    fi
+
+    echo -e "Kubeconfig file detected at: $OPTIONS_HUB_KUBECONFIG => copying to ./kube/config/hub-kubeconfig"
+    cp $OPTIONS_HUB_KUBECONFIG ./kube/config/hub-kubeconfig
+    export OPTIONS_HUB_KUBECONFIG=./kube/config/hub-kubeconfig
+
+    # Check to see if there are any kubecontext to be used from the hub cluster kubeconfig.
+    if [[ -z $OPTIONS_HUB_KUBECONTEXT || "$OPTIONS_HUB_KUBECONTEXT" == "null" ]]; then
+      HUB_CLUSTER=($(oc config get-clusters --kubeconfig=$OPTIONS_HUB_KUBECONFIG))
+      export OPTIONS_HUB_KUBECONTEXT=default/${HUB_CLUSTER[1]}/kube:admin
+    fi
+
+    echo -e
+    log_color "cyan" "Switching context to log into Kube API server"
+    oc config use-context --kubeconfig=$OPTIONS_HUB_KUBECONFIG $OPTIONS_HUB_KUBECONTEXT
+    
+    export OPTIONS_HUB_BASEDOMAIN=$(oc whoami --show-server=true | cut -d'.' -f2- | cut -d':' -f1)
+
+    if [[ $OPTIONS_HUB_OC_IDP == "kube:admin" && -z $OPTIONS_HUB_USER ]]; then
+      export OPTIONS_HUB_USER=kubeadmin
+    fi
+
+    log_color "purple" "HUB CLUSTER:" "$OPTIONS_HUB_BASEDOMAIN"
+  fi
+else
+  echo -e "Environment variables detected for hub cluster. Configuring tests to execute with exported variables."
 fi
 
+# Export variables to cypress.
+export CYPRESS_OPTIONS_HUB_BASEDOMAIN=$OPTIONS_HUB_BASEDOMAIN
+export CYPRESS_OPTIONS_HUB_KUBECONTEXT=$OPTIONS_HUB_KUBECONTEXT
+export CYPRESS_OPTIONS_HUB_PASSWORD=$OPTIONS_HUB_PASSWORD
+export CYPRESS_OPTIONS_HUB_USER=$OPTIONS_HUB_USER
+export CYPRESS_OPTIONS_HUB_OC_IDP=$OPTIONS_HUB_OC_IDP
+
+# Export base url for cluster.
+export BASE_URL=https://multicloud-console.apps.$OPTIONS_HUB_BASEDOMAIN
+export CYPRESS_BASE_URL=$BASE_URL
```

Comment:
Export variables for the hub cluster to cypress. This helps us to call the variables while running the UI test.